core/trino-main/src/main/java/io/trino/server/security/oauth2/OAuth2Service.java

```diff
@@ -128,10 +128,17 @@ public OAuthResult finishChallenge(Optional<UUID> authId, String code, URI callb
         // fetch access token
         AccessToken accessToken = client.getAccessToken(code, callbackUri);
 
+        String tok;
+        if (scopes.contains(OPENID_SCOPE)) {
+            tok = accessToken.getIdToken().get();
+        } else {
+            tok = accessToken.getAccessToken();
+        }
```

Comment:
Technically `scopes.contains(OPENID_SCOPE)` will always return `true` because we set it by default in `OAuth2Config` so if we keep that then we can update this to just use the ID token and no need for the extra conditional.